geoh5py/data/visual_parameters.py

```diff
@@ -0,0 +1,141 @@
+from .text_data import TextData
+
+PARAMETERS = ["Colour"]
+ATTRIBUTES = ["tag", "text", "attrib"]
+
+
+class VisualParameters(TextData):
+    _xml: ET.Element | None = None
+    _association = DataAssociationEnum.OBJECT
+
+    @property
+    def xml(self) -> ET.Element:
+        """
+        :obj:`str` XML string.
+        """
+        if self._xml is None:
+            if isinstance(self.values, str):
+                str_xml = self.values
+            else:
+                str_xml = """
+                    <IParameterList Version="1.0">
+                    </IParameterList>
+                """
+
+            self._xml = ET.fromstring(str_xml)
+
+        return self._xml
+
+    @property
+    def values(self):
+        if self._xml is not None:
+            self._values = ET.tostring(self._xml, encoding="unicode")
+
+        elif (getattr(self, "_values", None) is None) and self.on_file:
+            values = self.workspace.fetch_values(self)
+            if isinstance(values, (np.ndarray, str, type(None))):
+                self._values = values
+
+        return self._values
+
+    @values.setter
+    def values(self, values: np.ndarray | str | None):
+        self._values = values
+
+        if not isinstance(values, (np.ndarray, str, type(None))):
+            raise ValueError(
+                f"Input 'values' for {self} must be of type {np.ndarray}  str or None."
+            )
+
+        self.workspace.update_attribute(self, "values")
+
+    @property
+    def colour(self) -> None | list:
+        """
+        Colour of the object in [Alpha, Red, Green, Blue] format.
+
+        Each value is an integer between 0 and 255.
+        The colour value is stored as a single integer converted from
+        a byte string of the form 'BBGGRR00' where 'BB' is the blue value,
+        'GG' is the green value, 'RR' is the red converted from hexadecimal format.
+        """
+        element = self.get_tag("Colour")
+
+        if element is None or not element.text:
+            return None
+
+        c_string = (int(element.text)).to_bytes(4, byteorder="little").hex()
+
+        return [int(c_string[i : i + 2], 16) for i in range(0, 8, 2)]
+
+    @colour.setter
+    def colour(self, argb: list | tuple | np.ndarray):
+        if (
+            not isinstance(argb, (list, tuple, np.ndarray))
+            or len(argb) != 4
+            or not all(isinstance(val, int) for val in argb)
+        ):
+            raise TypeError("Input 'colour' values must be a list of 4 integers.")
+
+        byte_string = ""
+        for val in argb:
+            byte_string += f"{val:02x}"
+
+        value = int.from_bytes(bytes.fromhex(byte_string), "little")
+
+        self.set_tag("Colour", str(value))
+
+    def get_tag(self, tag: str) -> None | ET.Element:
+        """
+        Check if the tag is a valid parameter.
+        :param tag: the key of the tag to check.
+        :return: True if the tag is a valid parameter.
```

Comment:
**issue (doc):** wrong doc? Does not check the validity of the tag but returns its value.